vispy/app/backends/_qt.py

```diff
@@ -287,7 +287,12 @@ def __init__(self, *args, **kwargs):
         # Qt supports OS double-click events, so we set this here to
         # avoid double events
         self._double_click_supported = True
-        self._physical_size = p.size
+        if hasattr(self, 'devicePixelRatio'):
+            # handle high DPI displays in PyQt5
+            ratio = self.devicePixelRatio()
+        else:
+            ratio = 1
```

Comment:
This can probably just be `ratio = getattr(self, 'devicePixelRatio', lambda: 1.)()`

Reply:
Or `ratio = self.devicePixelRatio() if hasattr(self, 'devicePixelRatio') else 1.`. Your way isn't really my style (making a lambda as a default attribute).